mender.go

```diff
@@ -539,9 +566,61 @@ func reauthorize(m *mender) func() (client.AuthToken, error) {
 	}
 }
 
+// setNextServer is an iterator like function that cycles through the
+// list of available servers in mender.menderConfig.Servers and puts
+// the server as "current active", i.e. fills out the fields
+// mender.menderConfig.ServerURL and mender.menderConfig.TenantToken
+// with the corresponding fields in the server list.
+func setNextServer(m *mender) func() string {
+	if m.config.Servers == nil {
+		return nil
+	}
+
+	var skipIdx int
+	idx := 0
+	numServers := len(m.config.Servers)
+	// find index to skip
+	for i := 0; i < numServers; i++ {
+		if m.config.ServerURL == m.config.Servers[i].ServerURL {
+			skipIdx = i
+			break
+		}
+	}
+	return func() (srvURL string) {
+		if m.config.Servers == nil {
+			return ""
+		}
+		if idx == skipIdx {
+			idx++
+		}
+		// By taking modulo @numServers, the logic in ApiRequest.Do()
+		// defaults to the first Server in the list if all servers fails.
+		i := idx % numServers
+		// set server as current
+		m.config.ServerURL = m.config.Servers[i].ServerURL
+		m.config.TenantToken = m.config.Servers[i].TenantToken
+		if idx == numServers {
+			// We've tried all server, and reset to default server
+			return ""
+		} else {
+			idx++
+			// return serverURL with prefix (https?://) trimmed of
+			srv := m.config.Servers[i].ServerURL
+			_srv := strings.Split(srv, "://")
```

Comment:
strings.TrimPrefix?

Reply:
Just tried make it more generic.

Reply:
hb https://golang.org/pkg/net/url/ ?

Reply:
??

Reply:
Was just thinking it could be used for parsing the url, but might be a bit over the top.